src/room_member.cpp

```diff
@@ -0,0 +1,210 @@
+// Copyright 2017 Tunnler Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include <thread>
+#include <mutex>
+
+#include "tunnler/tunnler.h"
+#include "tunnler/room.h"
+#include "tunnler/room_member.h"
+#include "tunnler/room_message_types.h"
+
+#include "tunnler/assert.h"
+
+#include "BitStream.h"
+#include "RakNetTypes.h"
+
+RoomMember::RoomMember() {
+    peer = RakNet::RakPeerInterface::GetInstance();
+}
+
+RoomMember::~RoomMember() {
+    ASSERT_MSG(!IsConnected(), "RoomMember is being destroyed while connected");
```

Comment:
Why not just call Leave? Since this destructor my be called even if I'm connected (for example just closing citra while connected)